dockers/docker-database/docker-database-init.sh

```diff
@@ -22,6 +40,9 @@ then
 fi
 
 # generate all redis server supervisord configuration file
-sonic-cfggen -j /var/run/redis/sonic-db/database_config.json -t /usr/share/sonic/templates/supervisord.conf.j2 > /etc/supervisor/conf.d/supervisord.conf
+# Pass the Loopback IP explicitly in case of namespaces other than the linux host namespace.
+# This is needed as there are applications who hardcoded host=loopback IP in Connector classes.
+# Redis server will then bind on multiple IP addresses viz <loopback ip> , <eth0 ip>
+sonic-cfggen -j /var/run/redis/sonic-db/database_config.json -a "{\"LOOPBACK_IP\":\"$LOOPBACK_IP\"}" -t /usr/share/sonic/templates/supervisord.conf.j2 > /etc/supervisor/conf.d/supervisord.conf
```

Comment:
>$LOOPBACK_IP [](start = 84, length = 12)

I see `LOOPBACK_IP` will be undefined in some condition. Even if this is working, it is tricky and error prone. Could you improve?

Reply:
Thanks Qi for noting this, yes there will be case when LOOPBACK_IP variable is not used, in case of host namespace if [[ $NAMESPACE_ID == "" ]].  In this case the INTFC = "lo" and is already used to derived the loop back IP-address.

Since we pass the sonic-cfggen -a option with quotes "" it will go as empty string and  with **--bind {{ LOOPBACK_IP }} {{ redis_items['hostname'] }}**, it would just use what is defined in {{ redis_items['hostname'] }} alone.

Let me add a bit more comments to make it clear.

Reply:
To make it more clear I fine tuned a bit 

> If the host_ip derivation fails, set host_ip = 127.0.0.01, so it default to what is there now in database_config.json.
> moved the LOOPBACK check into supervisord.conf.j2. Here if redis_items['hostname'] is anything other than 127.0.0.1, we set LOOPBACK_IP and make it part of --bind option.